lib/friends/introvert.rb

```diff
@@ -203,7 +203,11 @@ def rename_location(old_name:, new_name:)
     # @param nickname [String] the nickname to add to the friend
     # @raise [FriendsError] if 0 or 2+ friends match the given name
     def add_nickname(name:, nickname:)
-      raise FriendsError, "Nickname cannot be blank" if nickname.empty?
+      if nickname.empty?
```

Comment:
Do we need these conditionals to be nested, or could we just do this?

```ruby
raise FriendsError, "Expected \"[Friend Name]\" \"[Nickname]\"" if name.empty?
raise FriendsError, "Nickname cannot be blank" if nickname.empty?
```

Reply:
Thanks for your comment!
I implemented like this because I thought it would emphasize that “Expected "[Friend Name]" "[Nickname]"” error is raised when neither of the two arguments are present.
However, this is just a subjective idea, so your idea may be better in terms of the more objective aspect of nesting depth.

Reply:
Thanks for explaining your thought process, @m-t-a-n-a-k-a! I have a preference for the change I suggested, in part because I actually think it's important that this code catches the widest possible error case.

This method shouldn't need to know that `name` is, in reality, always going be blank when `nickname` is also blank. If we later changed [the code that calls this](https://github.com/JacobEvelyn/friends/blob/main/lib/friends/commands/add.rb#L38), or added another call site, so that it could sometimes call:

```ruby
add_nickname(name: "", nickname: "not empty")
```

we'd still want to display an error because it's not clear what we'd want the method to do in that case.

Does that make sense?

Reply:
I get what you mean. I didnt realize that case.
Indeed, it's obvious when that case inserted and run and we compare the results.
```
$ cat lib/friends/commands/add.rb | grep "@introvert.add_nickname" 
      @introvert.add_nickname(name: "", nickname: "not empty")
      #@introvert.add_nickname(name: args.first.to_s.strip, nickname: args[1].to_s.strip)
```

```
#nested (my fix)
$ bundle exec ruby bin/friends add nickname
Error: No friend found for ""
```
```
#not nested (your idea)
$ bundle exec ruby bin/friends add nickname
Error: Expected "[Friend Name]" "[Nickname]"
```
I'll fix and push again later👍